components/base/kinematicbase/kinematics.go

```diff
@@ -202,3 +222,42 @@ func CollisionGeometry(cfg *referenceframe.LinkConfig) ([]spatialmath.Geometry,
 	}
 	return []spatialmath.Geometry{sphere}, nil
 }
+
+// newValidRegionCapsule returns a capsule which defines the valid regions for a base to be when moving to a waypoint.
+// The valid region is all points that are deviationThreshold (mm) distance away from the line segment between the
+// starting and ending waypoints. This capsule is used to detect whether a base leaves this region and has thus deviated
+// too far from its path.
+func newValidRegionCapsule(starting, desired []referenceframe.Input, deviationThreshold float64) (spatialmath.Geometry, error) {
+	pt := r3.Vector{X: (desired[0].Value + starting[0].Value) / 2, Y: (desired[1].Value + starting[1].Value) / 2}
+
+	capsule, err := spatialmath.NewCapsule(
+		spatialmath.NewZeroPose(),
+		deviationThreshold,
+		2*deviationThreshold+math.Sqrt(math.Pow(desired[0].Value-starting[0].Value, 2)+math.Pow(desired[1].Value-starting[1].Value, 2)),
+		"")
+	if err != nil {
+		return nil, err
+	}
+	// rotate such that y is forward direction to match the frame for movement of a base
+	o := &spatialmath.R4AA{
```

Comment:
Instead of using the zeroPose as the capsule Pose can just use NewPointFromOrientation and pass in the composition of the two orientations. Ideally we wouldn't need to do any transformations here since that makes extra work for us.